pthread/pthread.c

```diff
@@ -141,18 +141,11 @@ int pthread_join(pthread_t thread, void **value_ptr)
 	if (ctx->detached != PTHREAD_CREATE_DETACHED) {
 		int err;
 
-		do {
-			err = threadJoin(0);
+		err = threadJoin(ctx->id, 0);
 
-			if (err < 0) {
-				return err;
-			}
-			else if (ctx->id != err) {
-				pthread_ctx *ghost = find_pthread(err);
-				if (ghost != NULL)
-					ghost->detached = PTHREAD_CREATE_DETACHED;
-			}
-		} while (ctx->id != err);
+		if (err < 0) {
+			return err;
```

Comment:
We should check for -EINTR here - pthread_join is not supposed to return -EINTR, but threadJoin can do that. So we should loop util err == -EINTR